specifications/service-registry.yaml

```diff
@@ -0,0 +1,279 @@
+          description: longitude of the user location (WGS84)
+          schema:
+            type: number
+        - in: query
+          name: radius
+          description: Radius around the coordinates to search in kilometers
+          schema:
+            type: number
+        - in: query
+          name: certs
+          description:
+            Return only service providers owning the specified certificate
+            in their certificate chain
+          schema:
+            type: array
+            items:
+              type: string
+              format: uuid
+        - in: query
+          name: types
+          description: Return only services matching the specified types
+          style: form
+          explode: false
+          schema:
+            type: array
+            items:
+              type: string
+              enum:
+                - route-gtfs
+                - route-netex
+                - routing-otp
+                - taxi-suti
+                - tickets-lippu
+                - tickets-bob
+                - tickets-ticketapi
+            uniqueItems: true
+        - in: query
+          name: sort
+          description: Sort service providers in specified order
+          schema:
+            type: string
+            uniqueItems: true
+            enum:
+              - rating
+              - popularity
+              - alpha
+        - in: query
+          name: limit
+          description: Limit the maximum of the returned match to the specified number
+          schema:
+            type: integer
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/xml:
+              schema:
+                $ref: '#/components/schemas/service-catalog'
+            application/json:
+              schema:
+                $ref: '#/components/schemas/service-catalog'
+        '400':
+          description: Invalid query parameter
+        '403':
+          description: Access denied
+        '404':
+          description: Service not found
+
+components:
+  schemas:
+    catalog-listing:
+      type: object
+      properties:
+        catalogs:
+          type: array
+          items:
+            type: string
+            format: uri
+      example:
+        catalogs:
+          - 'https://tsp.example.com/catalog.json'
+          - 'https://maas.example.com/catalogs/services-20170101.json'
+    certificate:
+      type: object
+      required:
+        - id
+      properties:
+        id:
+          description:
+            Unique ID for the content of the certificate. The ID is the same for
+            all subjects of this certificate and can be used to query all
+            service providers who have received this certificate.
+          type: string
+          format: uuid
+          example: 75d2b5d5-75e9-4ae7-a60e-e58855fce6ea
+        type:
+          description: The type of the certificate, e.g. X.509
+          type: string
```

Comment:
Should this be an `enum`?

Reply:
Could this reuse what's in `ticket-api > identity > type`?

Reply:
It may be so that we don't limit the type of certificates that TSPs can use. Just a guess.

Reply:
My idea was just what @brylie said. I was thinking of letting the TSP and the user decide what certificates they can agree on. But I think there could be some better ways to harmonize the certificate and identity types between this, ticket-api and other APIs.